### ISSUE TITLE
Rockrobo v1 random map rotation

### ISSUE THREAD

**Issue by Crickus** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#303]
### Description

Is there any solution for random map 180 rotation at rockrobo v1 at every cleaning start? I know what it's MiHome problem, not extractor.

### Solution

Maybe it's possible to analyse image during extraction to auto rotate it? Some kind of test check "vacuum should start cleaning from the base situated in upper left corner" or something like this?
Or maybe someone already tried to do script/automatisation patch for this issue?

**Comment by PiotrMachowski** [comment]
Can you provide some examples of maps in different rotations? Is it the same as just rotating whole image?

**Comment by Crickus** [comment]
I almost sure what I saw topic about this issue at your v1 map repo, but with no solution.
This is the map image from your v1 project. It's cleared, but it present normal room direction. Vacuum station is in the upper left room.
![map_новый размер](https://user-images.githubusercontent.com/22008248/172929606-fc0ea462-964d-43a8-ad20-9c12e2500f1b.png)

Sometimes vacuum at cleaning start rotates map at 90 degrees. Today it rotated map at 180.
![Capture+_2022-06-09-22-22-44_новый размер](https://user-images.githubusercontent.com/22008248/172929834-6c9d10eb-3af5-43c6-88b5-35d8678ae3a9.png)

What's why I stopped using previous version. Map has changed randomly and coordinates have moved vacuum into wrong position.

As I understand it's some kind of Xiaomi defence algorithm to excludes more expensive models features from the cheapest one))

**Comment by PiotrMachowski** [comment]
Yes, there's no solution at this moment.

Can you provide raw images from Map Extractor instead? Ideally from config without map trimming

**Comment by t0bse** [comment]
got the same problem, the most times the roborock scans, the rotation is wrong.

**Comment by Crickus** [comment]
I pulled two images with different orientation today. It took three attempts (first and second was with the same orientation).
![map_image_rockrobo vacuum v1](https://user-images.githubusercontent.com/22008248/173133234-cc1fb847-8529-4105-8c95-af33eb4daa3f.png)
![map_image_rockrobo vacuum v1_reversed](https://user-images.githubusercontent.com/22008248/173133237-84f4c3cb-25db-4c44-835f-7a3a14606cc3.png)

**Comment by PiotrMachowski** [comment]
Can you change config of map extractor to following:
```yaml
camera:
  - platform: xiaomi_cloud_map_extractor
    host: !secret xiaomi_vacuum_host
    token: !secret xiaomi_vacuum_token
    username: !secret xiaomi_cloud_username
    password: !secret xiaomi_cloud_password
    draw: ['all']
    attributes:
      - calibration_points
      - image
```
and check out if this template correctly recognizes location of docking station?
```yaml
{% set docking = state_attr('camera.xiaomi_cloud_map_extractor', 'calibration_points')[0].map %}
{% set image = state_attr('camera.xiaomi_cloud_map_extractor', 'image').dimensions %}
{% set center = {'x': image.width * image.scale / 2, 'y': image.height * image.scale / 2, } %}
{{ ('upper' if docking.y <= center.y else 'lower') + '_' + ('left' if docking.x <= center.x else 'right') }}
```

**Comment by Crickus** [comment]
Yep. Your template recognized corner perfectly. If I understand your idea we need to wait until map changes it orientation and check it once again, right?
![map](https://user-images.githubusercontent.com/22008248/173146398-2e7bab25-a04e-4ba1-a5bc-6a38eb4615f0.jpg)
![template](https://user-images.githubusercontent.com/22008248/173146402-06bd5ffc-e9e6-4ec3-a67b-ca033a143d0c.jpg)

**Comment by PiotrMachowski** [comment]
> If I understand your idea we need to wait until map changes it orientation and check it once again, right?

Exactly. Also please check if CPU/RAM usage stayed at the same level

**Comment by Crickus** [comment]
The reversed map also recognized perfectly. CPU/RAM show more or less stable consumption (I have old Rasp3 HA server) .
![mem1](https://user-images.githubusercontent.com/22008248/173302854-bc55a41a-0d60-4d0c-8467-91146ca1ae57.jpg)
![proc1](https://user-images.githubusercontent.com/22008248/173302857-a00f18e2-25ac-47e6-9c60-34b283be9781.jpg)
![template_reverse](https://user-images.githubusercontent.com/22008248/173302859-beb617b0-02ba-4e30-b439-855acf48b84a.jpg)
One more detail. I'm not sure does vacuum rotate only the map or whole coordinates field. Even if we rotate the map in HA to normal direction, coordinates we transmit to API will be reversed, am I right?

**Comment by t0bse** [comment]
any news on this?

**Comment by LexaBukreev** [comment]
Could you please give an advice, how to enable room cleaning? I mapped rooms, but when i press start button (play button near x1 in a right side) - vacuum do nothing. Rooms a selectable, but after pressing start - it becomes unchecked and no response from vacuum. Xiaomi ver.1 P.S. I am ttrying to add an screenshot here, but "something went wrong"

**Comment by PiotrMachowski** [comment]
@LexaBukreev v1 vacuum doesn't support room cleaning
